torchvision/utils.py

```diff
@@ -428,7 +432,7 @@ def draw_keypoints(
                     width=width,
                 )
 
-    return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
+    return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=original_dtype)
```

Comment:
Just calling `.to` here won't scale the float images back down to `[0, 1]` so we would end up with a flaot image in `[0, 255]` (that's why the test would fail). It's best to just call `to_dtype()` for both the `uint8 <-> float` conversions.